grid2op/Opponent/RandomLineOpponent.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2019-2020, RTE (https://www.rte-france.com)
+# See AUTHORS.txt
+# This Source Code Form is subject to the terms of the Mozilla Public License, version 2.0.
+# If a copy of the Mozilla Public License, version 2.0 was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+import warnings
+import numpy as np
+
+from grid2op.Opponent import BaseOpponent
+from grid2op.Exceptions import OpponentError
+
+
+class RandomLineOpponent(BaseOpponent):
+    def __init__(self, action_space):
+        BaseOpponent.__init__(self, action_space)
+        self._do_nothing = None
+        self._attacks = None
+        self._lines_ids = None
+
+        # this is the constructor:
+        # it should have the exact same signature as here
+
+    def init(self, lines_attacked=[], **kwargs):
+        # this if the function used to properly set the object.
+        # It has the generic signature above,
+        # and it's way more flexible that the other one.
+
+        if len(lines_attacked) == 0:
+            warnings.warn(f'The opponent is deactivated as there is no information as to which line to attack. '
+                          f'You can set the argument "kwargs_opponent" to the list of the line names you want '
+                          f' the opponent to attack in the "make" function.')
+
+        # Store attackable lines IDs
+        self._lines_ids = []
+        for l_name in lines_attacked:
+            l_id = np.where(self.action_space.name_line == l_name)
+            if len(l_id) and len(l_id[0]):
+                self._lines_ids.append(l_id[0][0])
+            else:
+                raise OpponentError("Unable to find the powerline named \"{}\" on the grid. For "
+                                    "information, powerlines on the grid are : {}"
+                                    "".format(l_name, sorted(self.action_space.name_line)))
+
+        # Pre build do nothing action
+        self._do_nothing = self.action_space({})
```

Comment:
i think the "self._do_nothing" is not needed anymore

Reply:
Yes it's true